setup.py

```diff
@@ -79,17 +79,14 @@ def find_version(*file_paths):
     python_requires='>=3.6',
     install_requires=[
         "numpy",
-        "scipy",
-        "sparse",
-        "matplotlib",
         # Pinned due to https://github.com/pydata/sparse/issues/257
         # Ensure compatibility with numpy 1.17
         "numba>=0.46",
-        "scikit-image",
-        "libertem>=0.3.0"
     ],
     extras_require={
         'pyfftw': 'pyfftw',
+        'common': ['libertem>=0.4.0.dev0', 'scikit-image'],
+        'udf': ['libertem>=0.4.0.dev0', 'scikit-image', 'matplotlib'],
     },
     package_dir={"": "src"},
     packages=[
```

Comment:
Need to add the new packages here, i.e. `libertem_blobfinder.{base,common,udf}`. Other than that, it looks good! :+1:

Reply:
Thx, I am using `setuptools.find_namespace_packages` now. This should work, right? By the way, I didn't quite understand what effect that list has on the installation process. It worked fine with the incomplete list.Or is it just for information?

Reply:
> I am using setuptools.find_namespace_packages now. This should work, right?

**should**, yes :grin: it **should** be available in Python 3.3+, strange that it failed in 3.6. For some reason I still have some deep-seated distrust of `setuptools` functions like `find_packages`, that's why I didn't use them in the main LiberTEM package. If `find_packages` works fine, maybe we should adapt it for all packages.

> I didn't quite understand what effect that list has on the installation process. It worked fine with the incomplete list.Or is it just for information?

I am fairly certain that it at least *used* to be used for deciding what packages are actually part of the `sdist` and `wheel` - I have memories of needing to re-release because of missing complete packages. I could not reproduce this yet with the current build tooling, and it's been a few years...

Reply:
Hm, `tox` and CI **should** have failed then with the incomplete list of packages, if that was still a problem? It does build a wheel for the package and then installs it, right?

About mistrusting `find_packages`: We'd exchange us as error source with someone else's code as error source. 🤷‍♂ I can also go back to the explicit list if you prefer. :-) In any case, if there are problems we fix them.

Reply:
> Hm, tox and CI should have failed then with the incomplete list of packages, if that was still a problem? It does build a wheel for the package and then installs it, right?

`sdist` I think. The thing is, we also have a `MANIFEST.in` with a `graft src` statement (inherited from LiberTEM where it is needed for including the built client js files etc.) - this could have included the `.py` files "by accident" also.

> About mistrusting find_packages: We'd exchange us as error source with someone else's code as error source. 

Well, I think this mistrust is no longer deserved: it seems to work fine nowadays.

> I can also go back to the explicit list if you prefer. :-) In any case, if there are problems we fix them.

It seems to work, let's keep it like it is.